gen/gen.go

```diff
@@ -64,16 +70,16 @@ var (
 )
 
 // Struct generates Go structs based on the schema and writes them to w.
-func Struct(s string, w io.Writer, cfg Config) error {
+func Struct(s string, schemaMetadata *SchemaMetadata, w io.Writer, cfg Config) error {
```

Comment:
This is a breaking change. It would need to wait for a major version change, which will not happen now.